libs/dal/src/repositories/notification/notification.schema.ts

```diff
@@ -19,6 +19,7 @@ const notificationSchema = new Schema<NotificationDBModel>(
     _organizationId: {
       type: Schema.Types.ObjectId,
       ref: 'Organization',
+      index: true,
```

Comment:
The query for the stats is done with `_environmentId` and `_organizationId`. I would expect this field not being indexed, to slow down the query. Another option would be to just query based on the indexed field `_environmentId` as it should be enough.

Reply:
I have introduced the compound index of:
```
{ _environmentId: 1, createdAt: -1} 
```

yesterday, because of the cardinality of the environment field it will be better the the organizationId, and mongoDB is smart enough to use the new compound index for the new query even if it contains the organizationId.

One thing we can actually do is to remove the organizationId from the query.

Reply:
@scopsy The compound index was a great move. 👍🏻 
I think we should also introduce the index for organization for future multitenancy purposes. We might forget.